### PR TITLE
Upgrade plugin dependencies to React 18

### LINKED ISSUE #205: Upgrade plugin dependencies to React 18
As of Grafana10, Twinmaker runs in Grafana's React18 environment, so we should update React dev dependencies to 18 as well. 
Additionally, iot-app-kit with SceneViewer and VideoPlayer plugins have been upgraded to React18, so we should do this before we bump the versions. 

The [Grafana React 18 PR description](https://github.com/grafana/grafana/pull/64428) could be useful to see what needs to be upgraded

### PR DESCRIPTION
**Pull request by fridgepoet**
**What this PR does / why we need it**:
This upgrades plugin dependencies to React 18.

**Which issue(s) this PR fixes**:

Fixes #205

**Special notes for your reviewer**:
Context: https://raintank-corp.slack.com/archives/C01LVB0CYTW/p1692336771498409

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -26,18 +26,18 @@
     "@babel/core": "^7.16.7",
     "@emotion/css": "^11.1.3",
     "@grafana/aws-sdk": "0.1.2",
-    "@grafana/data": "8.4.11",
-    "@grafana/e2e": "8.4.11",
+    "@grafana/data": "10.0.3",
```
Comment (sarahzinger) [inline package.json:29]: Let's make sure we test this change with 9.4 for AMG, maybe we can reach out to folks at AWS to test on their end

File: src/panels/scene-viewer/SceneViewer.tsx [hunk 2]
```diff
@@ -227,11 +232,11 @@ export const SceneViewer = (props: SceneViewerPropsFromParent) => {
       ? (queryParams[getUrlTempVarName(props.options.customSelCompVarName)] as string)
       : undefined;
 
-    let selectedDataBindingResult: Record<string, string> | undefined =
+    let selectedDataBindingResult: ISelectedDataBinding | undefined =
       selectedEntityValue && selectedComponentValue
         ? {
-            [DataBindingLabelKeys.entityId]: selectedEntityValue,
```
Comment (sarahzinger) [inline src/panels/scene-viewer/SceneViewer.tsx:233]: Just curious why were these changes necessary?

**Review by sarahzinger — APPROVED** [review]
did a very brief test and the datasource seems to load ok for me! Let me know if there's anything in particular that would be good to test!

File: src/panels/scene-viewer/SceneViewer.tsx (hunk 2 above)
Comment (mukeshsahay) [reply src/panels/scene-viewer/SceneViewer.tsx:233]: This change is required due to the change in the definition for selectedDataBinding in iot-app-kit.

File: package.json [hunk 3]
```diff
@@ -118,6 +119,12 @@
     "debug": "^4.0.0",
     "underscore": "^1.12.1",
     "d3-color": "^3.1.0",
-    "log4js": "^6.4.0"
+    "log4js": "^6.4.0",
+    "react-dnd": "^16.0.1",
```
Comment (kevinwcyu) [inline package.json:123]: looks like `react-dnd@16.0.1` is the version that `iot-app-kit/*` v7 uses so i don't think we need the resolution here.

Comment (idastambuk) [reply package.json:123]: Good catch!

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,8 +1,13 @@
 # Changelog
 
+## 1.8.1
```
Comment (sarahzinger) [inline CHANGELOG.md:3]:
```suggestion
## 1.8.0
```

File: package.json [hunk 5]
```diff
@@ -1,9 +1,9 @@
 {
   "name": "grafana-iot-twinmaker-app",
-  "version": "1.7.1",
+  "version": "1.8.1",
```
Comment (sarahzinger) [inline package.json:3]:
```suggestion
  "version": "1.8.0",
```
